client/src/components/PhotosModal.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Photo from './Photo.jsx';
+import FontAwesome from '@fortawesome/fontawesome-free/js/all.js';
+
+export default class PhotosModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      roomId: null,
+      photoCount: 0,
+      photoArray: [],
+      currentPhoto: '',
+      currentPhotoIndex: 0,
+      nextVisibility: 'inline',
+      previousVisibility: 'none'
+    };
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    return {
+      roomId: props.roomId
+    };
+  }

```

Comment:
Can roomId be set in the constructor rather than just before render? I'm trying to think of the advantage here.